xapian-applications/omega/worker.cc

```diff
@@ -155,17 +155,23 @@ Worker::extract(const std::string & filename,
     string strpage;
 
     // Sending a filename and wating for the answer
-    if (write_string(sockt, filename) &&
-	read_string(sockt, dump) &&
-	read_string(sockt, title) &&
-	read_string(sockt, keywords) &&
-	read_string(sockt, author) &&
-	read_string(sockt, strpage)) {
-	    if (strpage.empty())
+    // If there is an error, the assistant process notify it through dump.
+    if (write_string(sockt, filename) && read_string(sockt, dump)) {
+	// Checks if the assistant succeded
+	if (dump == ASSISTANT_ERROR)
+	    return false;
```

Comment:
Using a magic value for a field like seems a rather awkward design as it can't distinguish an actual failure from a file containing the text `XAPIAN - ASSISTANT EXTRACTION FALSE`.  While that's not a big practical problem, it would be cleaner to simply change the protocol.  There's not compatibility worry at this point.

However, let's not try to address this in this PR and just continue to exit on error for now.

It's better to focus on getting what's here merged rather than try to include more changes to address other things which will just make the whole review process drag out even longer.  Several small PRs are typically easier to review than one PR including all the same changes.

Reply:
Okay, so Should I revert this change in a fix-up commit?